### ISSUE TITLE
[bug] When using multi-GPU / multi-node for pre-training, the initialized mm-projector weights & bias are different across each rank

### ISSUE THREAD

**Issue by hx-Tang** [haotian-liu/LLaVA#1440]
### Question

Thank you for your contribution :)

I notice this issue when I'm trying to reproduce your work and I found that there are some difference on the evaluation results of the models obtained using multi-node training and single-node training.

Then I dig in to the code and found that the newly initialized mm-projector weights & bias are different across each rank.

Insert this line before trainer.train() and I get different parameter accross ranks.

`
print(f"local_rank:{local_rank}, {[{i[0]:torch.mean(i[1].detach())} for i in trainer.model.get_model().mm_projector.named_parameters()]}")
`

e.g. for 4 node trainning I get:
`
local_rank:0, [{'0.weight': tensor(-6.6163e-06)}, {'0.bias': tensor(-0.0003)}, {'2.weight': tensor(-1.6939e-06)}, {'2.bias': tensor(-2.5525e-05)}]
local_rank:1, [{'0.weight': tensor(3.6777e-06)}, {'0.bias': tensor(-0.0004)}, {'2.weight': tensor(9.7775e-07)}, {'2.bias': tensor(-0.0002)}]
local_rank:2, [{'0.weight': tensor(-9.8941e-07)}, {'0.bias': tensor(-4.4031e-05)}, {'2.weight': tensor(-7.1693e-07)}, {'2.bias': tensor(0.0001)}]
local_rank:3, [{'0.weight': tensor(-8.1538e-06)}, {'0.bias': tensor(0.0003)}, {'2.weight': tensor(7.1939e-07)}, {'2.bias': tensor(4.7969e-05)}]
`

I try to set torch.manual_seed(seed) at the beginning of the train.py and this problem solved. 

I notice that there are many other LLaVA related work adopted your training code without reporting any issues on this, so I'm wondering whether I might be misunderstanding something, or if indeed there is a bug.

Thank you :)
